### DISCUSSION TITLE
ExtraNodesetGenerator with more nodes should be _id node - 1_

### DISCUSSION THREAD

**Discussion by RotmanC** [idaholab/moose#27884] [Q&A Meshing]
Hello.
I need to create a selection of nodes (more than 600), so I made this steps:

1. generate the mesh with gmsh
2. run the moose script with `--mesh-only`
3. open the `*_in.e` file it in paraview
4. copy the id nodes from the set of nodes (nodelist_7) by spreadsheet view
5. made `ExtraNodesetGenerator ` block into the `mesh` section
6. run again the moose script with `--mesh-only` to obtain the set of nodes

![image](https://github.com/idaholab/moose/assets/24434269/1ea0f1e7-f523-4b2e-b993-29fe60558e81)

Red dots are the set that moose gave after run the step 6.
Green light are the nodes that I want and I obtained after realize that the selection of nodes is: _id_node -1_

Nodes that I want to select and create the group of nodes (set)
```
Block_Name	Point_ID	ids	Points_0	Points_1	Points_2
nodelist_7		0	9734	5597.89		8118.03		-6000
nodelist_7		1	9737	5717.99		8407.98		-6000
nodelist_7		2	9745	5881.97		8675.57		-6000
nodelist_7		3	9770	8118.03		5597.89		-6000
nodelist_7		4	9773	7812.87		5524.62		-6000
nodelist_7		5	9781	7500		5500		-6000
```

Nodes that moose gave me after run step 6
```
Block_Name	Point_ID	ids		Points_0	Points_1	Points_2
new		0		9735		5301.25		8407.83		-6000
new		1		9738		5004.62		8697.63		-5875
new		2		9746		5712.52		9252.03		-5875
new		3		9771		8054.07		5188.86		-6000
new		4		9774		7990.1		4779.82		-5875
new		5		9782		7112.11		5026.31		-5875
```

Id's that I include in the `ExtraNodesetGenerator `  (_Id that I want - 1_)

```
9733
9736
9744
9769
9772
9780
```

It gives me the Id's that I want

![image](https://github.com/idaholab/moose/assets/24434269/7f280630-2eb4-4575-bfb9-1aa0677dea5f)
![image](https://github.com/idaholab/moose/assets/24434269/ec9ab29c-72cf-4765-a73a-b7d583616c80)

I didn't test with less selected nodes (e.g. to include a BC in an injection, the common selection is a single node or a few).

May be it could be specified in the [ExtraNodesetGenerator website](https://mooseframework.inl.gov/source/meshgenerators/ExtraNodesetGenerator.html) to avoid future issues.

**Comment by GiudGiud** [comment]
Hello

I think there's an offset of 1 between the numbering in paraview and the numbering in libmesh. This is a good suggestion though, I ll add a reminder there.

Guillaume

**Comment by RotmanC** [comment]
Thanks @GiudGiud 

By the way.... is there another option or methodology to do that?

All these steps were done because I need to assign a heat BC in a volume in a PorousFlow problem, so I created this group of nodes inside the volume that I want to....

**Reply by GiudGiud** [reply]
Hello

As long as these nodes / sides share a common geometrical characteristic, then there are plenty of ways to do this. 
One of the most general is to use the ParsedGenerateSideset and specify the equation that these sides satisfy.
If you need the nodes, there is a sideset to nodeset converter in moose

Guillaume

**Comment by RotmanC** [comment]
Thanks @GiudGiud !

I will consider it in future to avoid problems using different mesh management tools.

Rotman
